### DISCUSSION TITLE
Why is there a source range for virtual sources?

### DISCUSSION THREAD

**Discussion by vonglan** [helgoboss/helgobox#499] [Q&A] [answered]
For virtual targets, the target range is automatically hidden. Why isn't the source range for virtual sources hidden?

**Comment by helgoboss** [comment, marked as answer]
Absolute or relative control?

For relative control, because virtual targets don't have the concept of a current value, so restricting the target value range wouldn't be possible / make sense.

For absolute control, mmh, it wouldn't need a current target value to do the usual up/down-scaling to the target value range, so it would be possible. But I feel it wouldn't make sense. The result would be that (from the perspective of a main mapping with the corresponding virtual source) you would have a source that doesn't emit the full value range 0% to 100% but only a part of it. I don't see how this would be useful.

**Reply by helgoboss** [reply]
Ah, and source range makes sense. You might for example want to expose the first 50% of a knob turn as another virtual control element than the second 50%.

**Reply by vonglan** [reply]
> Ah, and source range makes sense. You might for example want to expose the first 50% of a knob turn as another virtual control element than the second 50%.

Makes sense, thanks!
